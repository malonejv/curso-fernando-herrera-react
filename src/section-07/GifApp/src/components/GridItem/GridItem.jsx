import React from 'react'
import './GridItem.css'
export const GridItem = ({ gif }) => {
  return (
    <div className='card'>
      <img src={gif.url} alt={gif.title} />
      <h3>{gif.title}</h3>
    </div>
  )
}
