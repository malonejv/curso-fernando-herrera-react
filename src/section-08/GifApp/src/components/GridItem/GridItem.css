h3 {
    font-size: 1rem;
    margin: 0.8rem 0 0 0;
    text-align: left;
    background-color: transparent;
    color: white;
}

img {
    background-color: rgb(51, 51, 51);
    border-radius: 8px;
}