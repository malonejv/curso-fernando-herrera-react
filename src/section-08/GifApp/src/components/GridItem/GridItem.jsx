import PropTypes from "prop-types";
import './GridItem.css';

export const GridItem = ({ gif }) => {
  return (
    <div className='card'>
      <img src={gif.url} alt={gif.title} />
      <h3>{gif.title}</h3>
    </div>
  )
}

GridItem.propTypes = {
  gif: PropTypes.shape({
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
  }).isRequired
}