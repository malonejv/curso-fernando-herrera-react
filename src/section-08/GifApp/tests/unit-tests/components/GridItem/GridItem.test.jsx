import { describe, test, expect } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import { GridItem } from "../../../../src/components/GridItem/GridItem";

describe("GridItem", () => {
  const defaultGif = {
    title: "Some title",
    url: "https://some-image.png/",
  };

  test("Should not render correctly with gif default object", () => {
    const gif = {};

    const { container } = render(<GridItem gif={gif} />);

    const h3Element = container.querySelector("h3");
    expect(h3Element.innerHtml).toBeUndefined();
  });

  test("Should not render correctly with gif empty", () => {
    const gif = {
      title: null,
      url: null,
    };

    const { container } = render(<GridItem gif={gif} />);
    const h3Element = container.querySelector("h3");
    expect(h3Element.innerHtml).toBeUndefined();
  });

  test("Should match snaptshot", () => {
    const { container } = render(<GridItem gif={defaultGif} />);
    expect(container).toMatchSnapshot();
  });

  test("Should render an image with url and title", () => {
    render(<GridItem gif={defaultGif} />);

    // const 
    //   src = screen.getByRole("img")?.src,
    //   alt = screen.getByRole("img")?.alt;
    const { src, alt } = screen.getByRole("img");

    expect(src).toEqual(defaultGif.url);
    expect(alt).toEqual(defaultGif.title);
  });
  
  test("Should render a title", ()=>{
    render(<GridItem gif={defaultGif} />);
    expect(screen.getByText(defaultGif.title)).toBeTruthy();
  });
});
