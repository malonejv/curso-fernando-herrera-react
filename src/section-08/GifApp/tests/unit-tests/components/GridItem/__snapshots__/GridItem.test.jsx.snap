// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem Should match snaptshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Some title"
      src="https://some-image.png/"
    />
    <h3>
      Some title
    </h3>
  </div>
</div>
`;
